packages/desktop-client/src/components/manager/BudgetList.tsx

```diff
@@ -239,7 +259,10 @@ function FileItem({
           )}
 
           {!quickSwitchMode && (
-            <FileMenuButton onDelete={() => onDelete(file)} />
+            <FileMenuButton
+              onDelete={() => onDelete(file)}
+              onDuplicate={'id' in file ? () => onDuplicate(file) : undefined}
```

Comment:
The management screen will list both cloud files that have been downloaded and cloud files that have not been downloaded.  Duplication will only work on local files (only has id) or cloud files that have been downloaded (have both id and cloudFileId).  If the file is only a cloudfile, then it will have a cloudFileId but won't have an id.
So, we need to check for the id as that determines if there is a local copy.  Later we also need the cloudfile logic to create a synced duplicate.

Logic is this:
- Local only file: Duplicate to only the local system. (User is not given a choice to create a synced file.)
- Cloud file synced on the local system: Give user choice to duplicate locally or both locally and on the server.
- Cloud file only (not synced): can not duplicate if it's not on the local system, so don't show the Duplicate menu item.

Reply:
Ah I see,. So to make this consistent whether server/local I guess it would need changes in actual-server. I suppose that can be added at another time.